sql/gcalc_tools.cc

```diff
@@ -161,26 +161,28 @@ int Gcalc_function::count_internal(const char *cur_func, uint set_type,
 
   if (next_func == op_border || next_func == op_internals)
   {
-    result= count_internal(cur_func,
-        (set_type == 1) ? set_type : next_func, &cur_func);
+    result= count_internal(cur_func, (set_type == 1) ? set_type : next_func,
+                           const_cast<const char **>(&cur_func));
```

Comment:
Why `const_cast`?

Reply:
I don't know. For some reason gcc failed to make this `const` by itself.

Reply:
Could you share how exactly?

Reply:
Here it is.
```c++
/work/mariadb/sql/gcalc_tools.cc:165:28: error: cannot initialize a parameter of type 'const char **' with an rvalue of type 'char **'
                           &cur_func);
                           ^~~~~~~~~
/work/mariadb/sql/gcalc_tools.cc:128:49: note: passing argument to parameter 'end' here
                                   const char **end)
                                                ^
1 error generated.
```